gp-includes/routes/glossary-entry.php

```diff
@@ -272,6 +274,14 @@ public function import_glossary_entries_post( $project_path, $locale_slug, $tran
 			return;
 		}
 
+		$replace = gp_post( 'import-replace' );
+		if ( 'on' === $replace && $this->can( 'write', 'project', $project->id ) ) {
+			$glossary_entries = GP::$glossary_entry->by_glossary_id( $glossary->id );
```

Comment:
Instead of looping through the glossary entries one by one and deleteing them you can just:

```
GP::$glossary_entry->delete_many( array( 'glossary_id', $glossary->id ) );
```